typescript/cli/src/config/submit.ts

```diff
@@ -57,6 +62,29 @@ export async function runSubmit({
   }
 }
 
+/**
+ * Retrieves the chain name from transactions[0].
+ *
+ * @param multiProvider - The MultiProvider instance to use for chain name lookup.
+ * @param transactions - The list of populated transactions.
+ * @returns The name of the chain that the transactions are submitted on.
+ * @throws If the transactions are not all on the same chain or chain is not found
+ */
+function getChainFromTxs(
+  multiProvider: MultiProvider,
+  transactions: PopulatedTransactions,
+) {
+  const firstTransaction = transactions.at(0);
+  assert(firstTransaction, 'Transactions empty');
+
+  const sameChainIds = transactions.every(
+    (t: PopulatedTransaction) => t.chainId === firstTransaction.chainId,
+  );
+  assert(sameChainIds, 'Transactions must be submitted on the same chains');
```

Comment:
interesting, I guess we could use this as the means of multichain support too?

Reply:
I will remove this when we allow multichain support in `hyperlane submit`. it's currently restricted to a single chain anyways